### ISSUE TITLE
Support prerelease versions (e.g. as used by cargo-release)

### ISSUE THREAD

**Issue by tarcieri** [RustRPM/cargo-rpm#3]
Moving this over from:

https://github.com/iqlusion-io/crates/issues/38

The version format used for cargo prereleases does not seem to be supported by RPM. This results in failures like:

```
$ cargo rpm build
    Finished release [optimized] target(s) in 0.0 secs
    Building sakcl-0.1.1-alpha.0.rpm (using rpmbuild 4.11.3)

error: error running /usr/bin/rpmbuild (exit status: exit code: 1)
ERROR: Job failed: exit code 1
```

cc @cardoe

**Comment by sharksforarms** [comment]
Having the same issue

**Comment by tarcieri** [comment]
This seems easy enough to fix if someone can suggest a way to munge cargo prerelease versions into a format RPM would find acceptable.

Ideally, IMO, we could find a format both find mutually acceptable, and enforce use of that convention with `cargo-rpm`

**Comment by sharksforarms** [comment]
What if we put it in the RPM `Release` field?
i.e.
```
Version: 0.1.1
Release: alpha.0
```

**Comment by tarcieri** [comment]
Sounds good to me. Mind testing that out and seeing if it works?

**Comment by cardoe** [comment]
Unfortunately that won’t be valid. For information about what’s valid in RPM Version and Release fields see: https://fedoraproject.org/wiki/Packaging:Versioning

**Comment by tarcieri** [comment]
@cardoe thanks! It seems this is what we have to work with:

> In the `Version:` tag, use the version that upstream has determined the next release will be. For the `<pkgrel>` field of the `Release:` tag, use a number of the form `"0.N"` where `N` is an integer beginning with `1` and increasing for each revision of the package. Prerelease versions MUST use a `Release:` tag strictly less than `1`, as this is the sole indicator that a prerelease has been packaged.

So it seems some sort of munging is required to go from prerelease versions supported by Cargo to a purely numerical prerelease identifier.

**Comment by sharksforarms** [comment]
I tested it and it does let me put `Release: alpha.0`, the RPM that it generates looks like `helloworld-1.0.1-alpha.0.x86_64.rpm` but it;s probably not the best solution as others have mentioned.

Also found this this:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html-single/rpm_packaging_guide/index
> The number of times this version of the software was released. Normally, set the initial value to 1%{?dist}, and increment it with each new release of the package. Reset to 1 when a new Version of the software is built. 

To get a feel of the formats out there
`rpm -qa | xargs -i rpm -q --queryformat '%{Release}\n' {}`

Some examples:
https://fedoraproject.org/wiki/Package_Versioning_Examples

Might be enough to flip the schema around?
`alpha.0` -> `0.alpha%{?dist}`

**Comment by tarcieri** [comment]
I asked the [Fedora Rust SIG](https://docs.pagure.org/fedora-rust.sig/index.html) for input. We'll see what they say.

> Might be enough to flip the schema around?

So concretely would that be...

- `alpha.0` -> `0.alpha.0`
- `alpha.1` -> `0.alpha.1`

?

Alternatively I can see mandating a particular label for prereleases (e.g. `pre`) and dropping that from the `Release`, e.g.:

- `1.0.0-pre1` -> Version: `1.0.0`, Release: `0.1`
- `1.0.0-pre2` -> ... Release `0.2`

**Comment by tarcieri** [comment]
Related rust2rpm issue: https://pagure.io/fedora-rust/rust2rpm/issue/3

**Comment by tarcieri** [comment]
Support for this is released in v0.6.0. Thanks @jcreekmore!
